src/EFCore.Cosmos/Query/Internal/CosmosSqlTranslatingExpressionVisitor.cs

```diff
@@ -820,18 +820,28 @@ protected override Expression VisitUnary(UnaryExpression unaryExpression)
             ExpressionType.Negate or ExpressionType.NegateChecked
                 => sqlExpressionFactory.Negate(sqlOperand!),
 
-            ExpressionType.Convert or ExpressionType.ConvertChecked
-                when operand.Type.IsInterface
-                && unaryExpression.Type.GetInterfaces().Any(e => e == operand.Type)
+            // Convert nodes can be an explicit user gesture in the query, or they may get introduced by the compiler (e.g. when a Child is
+            // passed as an argument for a parameter of type Parent). The latter type should generally get stripped out as a pure C#/LINQ
+            // artifact that shouldn't affect translation, but the latter may be an indication from the user that they want to apply a
+            // type change.
+            ExpressionType.Convert or ExpressionType.ConvertChecked or ExpressionType.TypeAs
+                when operand.Type.IsInterface && unaryExpression.Type.GetInterfaces().Any(e => e == operand.Type)
+                // We strip out implicit conversions, e.g. float[] -> ReadOnlyMemory<float> (for vector search)
+                || (unaryExpression.Method is { IsSpecialName: true, Name: "op_Implicit" }
+                    && IsRom(unaryExpression.Type.UnwrapNullableType()))
```

Comment:
Ah so without this added check tests were failing? Interesting...

I do think removing all implicit casts should probably make sense - without the check specifically for the ROM here (since it's an artifact of the compiler rather than something with any actual meaning). But we can explore later as needed.

(though nit-wise I'd spell out ReadOnlyMemory:)

```suggestion
                    && IsReadOnlyMemory(unaryExpression.Type.UnwrapNullableType()))
```